Tests/Unit/Model/SepaDirectDebitPaymentMethodTest.php

```diff
@@ -0,0 +1,129 @@
+use Wirecard\Oxid\Model\Transaction;
+
+use OxidEsales\Eshop\Core\Field;
+use OxidEsales\Eshop\Core\Registry;
+
+use Wirecard\PaymentSdk\Transaction\SepaDirectDebitTransaction;
+
+class SepaDirectDebitPaymentMethodTest extends OxidEsales\TestingLibrary\UnitTestCase
+{
+    /**
+     * @var SepaDirectDebitPaymentMethod
+     */
+    private $_oPaymentMethod;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->_oPaymentMethod = new SepaDirectDebitPaymentMethod();
+    }
+
+    public function testGetConfig()
+    {
+        $oConfig = $this->_oPaymentMethod->getConfig();
+        $this->assertNotNull($oConfig);
+    }
+
+    public function testGetConfigFields()
+    {
+        $aConfigFields = $this->_oPaymentMethod->getConfigFields();
+        $this->assertArrayHasKey("sepaMandateCustom", $aConfigFields);
+        $this->assertArrayHasKey("creditorId", $aConfigFields);
+        $this->assertArrayHasKey("additionalInfo", $aConfigFields);
+        $this->assertArrayHasKey("bic", $aConfigFields);
+    }
+
+    public function getDefaultConfigFieldsProvider()
+    {
+        return [
+            "contains apiUrl"=> ['apiUrl'],
+            "contains httpUser"=> ['httpUser'],
+            "contains httpPassword"=> ['httpPassword'],
+            "contains maid"=> ['maid'],
+            "contains secret"=> ['secret'],
+            "contains testCredentials"=> ['testCredentials'],
+        ];
+    }
+
+    public function testGetCheckoutFields()
+    {
+        $oPayment = $this->_oPaymentMethod->getPayment();
+        $oPayment->oxpayments__wdoxidee_bic->value = new Field(1);
+        $oPayment->save();
+
+        $aPublicFieldNames = $this->_oPaymentMethod->getCheckoutFields();
+        $aExpected = [
+            "accountHolder",
+            "iban",
+            "bic",
+        ];
+
+        $this->assertEquals($aExpected, $aPublicFieldNames, '', 0.0, 1, true);
+    }
+
+    public function testGetTransaction()
+    {
+        $oTransaction = $this->_oPaymentMethod->getTransaction();
+        $this->assertInstanceOf(SepaDirectDebitTransaction::class, $oTransaction);
+    }
+
+    public function testAddMandatoryTransactionData()
+    {
+        $oTransaction = $this->_oPaymentMethod->getTransaction();
+        $aDynArray = Registry::getSession()->getVariable("dynvalue");
+        if (!$aDynArray) {
+            $aDynArray = [];
+        }
+        $aDynArray['bic'] = 'WIREDEMMXXX';
+        Registry::getSession()->setVariable('dynvalue', $aDynArray);
+        $this->_oPaymentMethod->addMandatoryTransactionData($oTransaction);
+        $this->assertAttributeEquals('', 'iban', $oTransaction);
+        $this->assertAttributeNotEmpty('mandate', $oTransaction);
+        $this->assertAttributeNotEmpty('accountHolder', $oTransaction);
+    }
+
+    public function testGetPublicFieldNames()
+    {
+        $aPublicFieldNames = $this->_oPaymentMethod->getPublicFieldNames();
+        $this->assertCount(7, $aPublicFieldNames);
```

Comment:
This assertion is not needed because of the `assertEquals()` below.

Reply:
👍